GAME/TRAPS.C

```diff
@@ -122,8 +123,18 @@ void ControlRollingBall(short item_number)//5AE08, 5B284
 
 void LavaBurn(struct ITEM_INFO* item)//5AD78, 5B1F4
 {
-	S_Warn("[LavaBurn] - Unimplemented!\n");
-	return;
+	if (item->hit_points >= 0 && lara.water_status != 3)
+	{
+		short room_number = item->room_number;
```

Comment:
All variables must be declared at the start of a method.

Reply:
C89 states that variables must be declared at the start of a scope block, which is the case here